src/Application/Auth/PersonManagementAuthMiddleware.php

```diff
@@ -37,18 +43,26 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
         }
 
         $personId = $route->getArgument('personId');
-        $jws = $request->getHeaderLine('x-tbg-auth');
+        $this->jws = $request->getHeaderLine('x-tbg-auth');
 
-        if (empty($jws)) {
+        if (empty($this->jws)) {
             $this->logger->info('Security: No JWT provided');
             $this->unauthorised($this->logger, true, $request);
         }
 
-        if (!$this->token->check($personId, $jws, $this->logger)) {
+        if (!$this->token->check($personId, $this->jws, $this->logger)) {
             $this->logger->info(sprintf('Security: JWT check failed for person ID %s', $personId));
             $this->unauthorised($this->logger, false, $request);
         }
 
-        return $handler->handle($request->withAttribute('pspId', IdentityToken::getPspId($jws)));
+        $this->checkCompleteness($request);
+
+        return $handler->handle($request->withAttribute('pspId', IdentityToken::getPspId($this->jws)));
+    }
+
+    protected function checkCompleteness(ServerRequestInterface $request): void
+    {
+        // No-op: we allow both values for this base middleware. This should be extended
+        // by middlewares that have tighter requirements.
```

Comment:
@NoelLH If that's the case, then perhaps we should make this middleware class an abstract class?

Reply:
@alihejazi This cannot be abstract because it is invoked directly too, for `Donations\Create`.

The idea is that this general version of the middleware is less constrained, but its subclasses can be more so.

If we made it abstract and extended it for `Donations\Create`, we would still have to make the subclass implement the method as a no-op, so as I see it that would be more complexity and an extra class for no real benefit.